src/utils/TokenUtils.js

```diff
@@ -211,14 +210,10 @@ define(function (require, exports, module) {
      * @return {mode:{Object}, name:string}
      */
     function getModeAt(cm, pos) {
-        var outerMode = cm.getMode(),
-            modeData = CodeMirror.innerMode(outerMode, cm.getTokenAt(pos, true).state),
```

Comment:
The only thing that I don't understand, is what happened to the second argument of `getTokenAt`. This seemed to be something that cm refers to as `precision`. This is now handled differently in cm. It seems, following `cm.getModeAt` code path, it will be undefined. Perhaps this doesn't make any difference, but I don't know of a good way to proof it. Perhaps @RaymondLim knows more about the internal mechanics of cm.